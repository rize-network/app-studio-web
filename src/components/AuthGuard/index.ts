export { AuthGuard, withAuthGuard, RouteAuthGuard, AuthStatus } from './AuthGuard';
