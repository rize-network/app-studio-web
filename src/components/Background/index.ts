export { Background } from './Background';
export type {
  BackgroundProps,
  AuroraBackgroundProps,
  MeteorsProps,
  WallProps,
  ParticlesProps,
  GridProps,
  RipplesProps,
  BackgroundImageProps,
  BackgroundVideoProps,
  BackgroundGradientProps,
  BackgroundStyles,
} from './Background/Background.props';
export type {
  BackgroundEffect,
  AuroraDirection,
  MeteorTiming,
  BorderAnimationStyles,
  StrokeAnimationStyles,
  BackgroundContextType,
  BackgroundSize,
  BackgroundPosition,
  BackgroundRepeat,
  BlendMode,
} from './Background/Background.type';
