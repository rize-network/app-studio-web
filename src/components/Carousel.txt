'use client';
import React, { useEffect, useRef, useState, createContext } from 'react';
import { Horizontal, Vertical } from './Layout';
import { IconButtons } from './Button/examples';
import { Button } from './Button/Button';
import { RightArrowSvg } from 'src/components/Svg/index';

// import { IconLeft, IconRight } from '@app-studio/web/icons';

interface CarouselProps {
  items: JSX.Element[];
  initialScroll?: number;
}

export const CarouselContext = createContext({
  onCardClose: (index: number) => {},
  currentIndex: 0,
});

export const Carousel = ({ items, initialScroll = 0 }: CarouselProps) => {
  const carouselRef = useRef<HTMLDivElement>(null);
  const [canScrollLeft, setCanScrollLeft] = useState(false);
  const [canScrollRight, setCanScrollRight] = useState(true);

  useEffect(() => {
    if (carouselRef.current) {
      carouselRef.current.scrollLeft = initialScroll;
      checkScrollability();
    }
  }, [initialScroll]);

  const checkScrollability = () => {
    if (carouselRef.current) {
      const { scrollLeft, scrollWidth, clientWidth } = carouselRef.current;
      setCanScrollLeft(scrollLeft > 0);
      setCanScrollRight(scrollLeft < scrollWidth - clientWidth);
    }
  };

  const scrollLeft = () =>
    carouselRef.current?.scrollBy({ left: -300, behavior: 'smooth' });
  const scrollRight = () =>
    carouselRef.current?.scrollBy({ left: 300, behavior: 'smooth' });

  return (
    <CarouselContext.Provider
      value={{ onCardClose: () => {}, currentIndex: 0 }}
    >
      <Horizontal
        gap={2}
        alignItems="center"
        width="100%"
        paddingVertical={20}
        position="relative"
      >
        <Button
          onClick={scrollLeft}
          icon={<RightArrowSvg />}
          disabled={!canScrollLeft}
          shape="rounded"
        />
        <Horizontal
          ref={carouselRef}
          onScroll={checkScrollability}
          gap={4}
          overflowX="auto"
          paddingX={10}
        >
          {items.map((item, index) => (
            <Vertical key={index} minWidth="300px">
              {item}
            </Vertical>
          ))}
        </Horizontal>
        <Button
          onClick={scrollRight}
          icon={<RightArrowSvg />}
          disabled={!canScrollRight}
          shape="rounded"
        />
      </Horizontal>
    </CarouselContext.Provider>
  );
};
