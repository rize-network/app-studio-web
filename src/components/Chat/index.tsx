/**
 * App Studio Components for AI Applications
 *
 * This module exports a collection of components specifically designed
 * for building AI-powered applications.
 */

// Core components
export * from './ChatInterface/ChatInterface';
export * from './MessageBubble/MessageBubble';
export * from './PromptInput/PromptInput';
export * from './ModelSelector/ModelSelector';
export * from './AIResponseDisplay/AIResponseDisplay';
export * from './LoadingIndicator/LoadingIndicator';

// Supporting components
export * from './ToolSelector/ToolSelector';
export * from './SettingsPanel/SettingsPanel';
export * from './FileUploader/FileUploader';
export * from './CodeBlock/CodeBlock';
