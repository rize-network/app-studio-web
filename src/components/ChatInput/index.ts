export { ChatInput } from './ChatInput';
export type { ChatInputHandles } from './ChatInput';
export { AttachmentGroup } from './AttachmentGroup';
export { EditableInput } from './EditableInput';
export { FileUploadHandler } from './FileUploadHandler';
export { GuideTip } from './GuideTip';
export { PromptExamples } from './PromptExamples';
export { ReferenceImageButton } from './ReferenceImageButton';
export { ReferenceImageModal } from './ReferenceImageModal';
