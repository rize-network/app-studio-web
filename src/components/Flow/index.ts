export * from './Flow';
export * from './examples/DragAndDrop';
