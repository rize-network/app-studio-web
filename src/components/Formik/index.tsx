export { FormikCheckbox } from './Formik.Checkbox';
export { FormikDatePicker } from './Formik.DatePicker';
export { FormikCountryPicker } from './Formik.CountryPicker';
export { FormikSelect } from './Formik.Select';
export { FormikSwitch } from './Formik.Switch';
export { FormikTextArea } from './Formik.TextArea';
export { FormikTextField } from './Formik.TextField';
export { FormikPassword } from './Formik.Password';
export { FormikForm } from './Formik.Form';
export { FormikComboBox } from './Formik.ComboBox';
export { FormikSlider } from './Formik.Slider';
