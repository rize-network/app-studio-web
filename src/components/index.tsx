export * from './Button/Button';
export * from './Form/Checkbox/Checkbox';
export * from './Form/CountryPicker/CountryPicker';
export * from './Form/DatePicker/DatePicker';
export * from './Form/Password/Password';
export * from './Form/Select/Select';
export * from './Form/Switch/Switch';
export * from './Form/Switch/Switch';
export * from './Form/TextArea/TextArea';
export * from './Form/TextField/TextField';
export * from './Layout/Center/Center';
export * from './Formik';
export * from './Layout/Horizontal/Horizontal';
export * from './Layout/Vertical/Vertical';
export * from './Layout/View/View';
export * from './Link/Link';
export * from './Loader/Loader';
export * from './Modal/Modal';
export * from './Text/Text';
export * from './File/File';
export * from './Tabs/Tabs';
export * from './Alert/Alert';
export * from './AspectRatio/AspectRatio';
export * from './Avatar/Avatar';
export * from './Badge/Badge';
export * from './Toggle/Toggle';
export * from './ToggleGroup/ToggleGroup';
export * from './Message/Message';
