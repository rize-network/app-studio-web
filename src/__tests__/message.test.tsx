import React from 'react';
import renderer from 'react-test-renderer';
import { Button, View, showMessage, MessageLayout } from '../components';

import { cleanup } from '@testing-library/react';

afterEach(() => {
  cleanup();
});

test('AspectRatio to match snapshot', () => {
  const tree = renderer
    .create(
      <View>
        <Button
          onClick={() =>
            showMessage(
              'success',
              'Scheduled: Catch up',
              'Friday, February 10, 2023 at 5:57 PM'
            )
          }
        >
          Show Toast
        </Button>
        <MessageLayout />
      </View>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
