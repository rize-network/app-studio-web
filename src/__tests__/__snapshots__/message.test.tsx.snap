// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AspectRatio to match snapshot 1`] = `
<div
  className="sc-bdfCDU cYBAia"
>
  <button
    className="sc-bdfCDU bjTbpr"
    color="color.white"
    cursor="pointer"
    disabled={false}
    display="flex"
    filter="brightness(1)"
    fontSize="md"
    fontWeight={600}
    letterSpacing={1.25}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="button"
    transform=""
    width={144}
  >
    <div
      className="sc-bdfCDU dpBfNq"
      display="flex"
    >
      Show Toast
    </div>
  </button>
</div>
`;
